shared/actions/login/saga.js

```diff
@@ -403,14 +403,25 @@ function addDeviceRpc (channelConfig) {
   return Types.deviceDeviceAddRpcChannelMapOld(channelConfig, {})
 }
 
+function * handleProvisioningError (error) {
+  yield put(navigateAppend([{
+    props: {
+      error,
+    },
+    selected: 'error',
+  }], [loginTab, 'login']))
+  yield take(Constants.onBack)
```

Comment:
The right way here is to have the error screen dispatch a different action that causes a saga to run `navBasedOnLoginState`. This was just easier